cwltest/cwl_junit_xml.py

```diff
@@ -0,0 +1,157 @@
+
+        if self.hostname:
+            test_suite_attributes['hostname'] = decode(self.hostname, encoding)
+        if self.id:
+            test_suite_attributes['id'] = decode(self.id, encoding)
+        if self.package:
+            test_suite_attributes['package'] = decode(self.package, encoding)
+        if self.timestamp:
+            test_suite_attributes['timestamp'] = decode(self.timestamp, encoding)
+        if self.timestamp:
+            test_suite_attributes['file'] = decode(self.file, encoding)
+        if self.timestamp:
+            test_suite_attributes['log'] = decode(self.log, encoding)
+        if self.timestamp:
+            test_suite_attributes['url'] = decode(self.url, encoding)
+
+        xml_element = ET.Element("testsuite", test_suite_attributes)
+
+        # add any properties
+        if self.properties:
+            props_element = ET.SubElement(xml_element, "properties")
+            for k, v in self.properties.items():
+                attrs = {'name': decode(k, encoding), 'value': decode(v, encoding)}
+                ET.SubElement(props_element, "property", attrs)
+
+        # add test suite stdout
+        if self.stdout:
+            stdout_element = ET.SubElement(xml_element, "system-out")
+            stdout_element.text = decode(self.stdout, encoding)
+
+        # add test suite stderr
+        if self.stderr:
+            stderr_element = ET.SubElement(xml_element, "system-err")
+            stderr_element.text = decode(self.stderr, encoding)
+
+        # test cases
+        for case in self.test_cases:
+            test_case_attributes = dict()
+            test_case_attributes['name'] = decode(case.name, encoding)
+            if case.assertions:
+                # Number of assertions in the test case
+                test_case_attributes['assertions'] = "%d" % case.assertions
+            if case.elapsed_sec:
+                test_case_attributes['time'] = "%f" % case.elapsed_sec
+            if case.timestamp:
+                test_case_attributes['timestamp'] = decode(case.timestamp, encoding)
+            if case.classname:
+                test_case_attributes['classname'] = decode(case.classname, encoding)
+            if case.status:
+                test_case_attributes['status'] = decode(case.status, encoding)
+            if case.category:
+                test_case_attributes['class'] = decode(case.category, encoding)
+            if case.file:
+                test_case_attributes['file'] = decode(case.file, encoding)
+            if case.line:
+                test_case_attributes['line'] = decode(case.line, encoding)
+            if case.log:
+                test_case_attributes['log'] = decode(case.log, encoding)
+            if case.url:
+                test_case_attributes['url'] = decode(case.url, encoding)
+            if case.short_name:
+                test_case_attributes['short_name'] = decode(case.short_name, encoding)
```

Comment:
We can't just add new attributes, Jenkins will ignore them or crash and this creates a lot of maintenance burden. Lets find a better place to put the short name.

Reply:
I used `classname`attribute